cmd/routed-eni-cni-plugin/cni.go

```diff
@@ -192,16 +204,20 @@ func add(args *skel.CmdArgs, cniTypes typeswrapper.CNITYPES, grpcClient grpcwrap
 	}
 
 	var hostVethName string
+	var dummyVlanInterface *current.Interface
 	if r.PodVlanId != 0 {
-		hostVethName = generateHostVethName("vlan", string(k8sArgs.K8S_POD_NAMESPACE), string(k8sArgs.K8S_POD_NAME))
-
+		hostVethName = generateHostVethName(vlanInterfacePrefix, string(k8sArgs.K8S_POD_NAMESPACE), string(k8sArgs.K8S_POD_NAME))
 		err = driverClient.SetupPodENINetwork(hostVethName, args.IfName, args.Netns, v4Addr, v6Addr, int(r.PodVlanId), r.PodENIMAC,
 			r.PodENISubnetGW, int(r.ParentIfIndex), mtu, log)
+		// This is a dummyVlanInterface is created only to send podVlanId information as a part of the Result struct
+		// The podVlanId is used by DEL cmd fetched from the prevResult struct to cleanup pod network
+		dummyVlanInterfaceName := generateHostVethName(dummyVlanInterfacePrefix, string(k8sArgs.K8S_POD_NAMESPACE), string(k8sArgs.K8S_POD_NAME))
```

Comment:
As @achevuru mentioned, let's document on why we added the dummyVlanInterfaceName for future reference.

Reply:
Updated comments